### DISCUSSION TITLE
Using Seurat V5 with non-10x genomics datasets

### DISCUSSION THREAD

**Discussion by Itsyourgurlmurl** [satijalab/seurat#9175] [Q&A]
Hi everyone, 
I am so sorry if this question has been asked a lot already, but I have not yet been able to find the answer (I am quite new to this):
Since the update Seurat underwent to V5, is it still possible to use that for datasets that are non-10x Genomics? 
My dataset worked fine with V4 but I can't even read in my count tables using V5.. For a while, I happily kept using V4, but now I want to use some analyses for which V5 is required. And of course the million dollar question: If possible, how can I modify my Counttables to be suitable input for V5?
Thanks so much in advance for any input!

**Comment by samuel-marsh** [comment]
Hi,

the only thing needed to create Seurat object is data where rows are genes and columns are cells.  You can provide this data.frame or matrix to CreateSeuratObject.  It shouldn’t be different than in prior versions of Seurat.

Best,
Sam

**Reply by Itsyourgurlmurl** [reply]
Hi Sam, when I load in my counttables I get the following error message:

> countTablesFiltered.RData <- Read10X(data.dir = "C:/Users/3750221/Desktop/RNAseq_analysis/Counttables_extracted/")
Error in Read10X(data.dir = "C:/Users/3750221/Desktop/RNAseq_analysis/Counttables_extracted/") : 
  **Barcode file missing. Expecting barcodes.tsv.gz**
  
_When I try to run it with the same code as I used in V4 it will load in but then I get the following problem:_ 

plateDescriptions <- c("WT pure (H3K4me3)", "CRC pure (H3K4me3)", "WT pure (H3K27me3)", "CRC pure (H3K27me3)",
                       "WT mix (H3K4me3)", "CRC mix (H3K4me3)", "WT mix (H3K27me3)", "CRC mix (H3K27me3)")
scPlates <- list.files("Counttables_extracted", include.dirs = TRUE, full.names = TRUE, recursive = TRUE, 
                       pattern = "RNA_gene_counts.csv") 
filePrefix <- list.files("Counttables_extracted", pattern = "SCC-")
countTables <- readRDS("countTablesFiltered.RData")
objNames <- paste(filePrefix, "Obj", sep = "")
objNames <- gsub("-", "", objNames)
for (i in 1:length(scPlates)){
  seuratObj <- CreateSeuratObject(counts = countTables[[i]],
                                  min.cells = 3, 
                                  min.features = 100,
                                  project = plateDescriptions[i])
  assign(objNames[i], seuratObj)
}
**Warning: Data is of class data.frame. Coercing to dgCMatrix.**

_So I then converted the data.frame to dgCMatrix:_

sapply(countTables, class)
library(Matrix)
convert_to_sparse <- function(df){
  non_zero_indices <- which(df != 0, arr.ind = TRUE)
  sparseMatrix(
    i = non_zero_indices[, 1],
    j = non_zero_indices[, 2],
    x = df[non_zero_indices],
    dims = dim(df),
    dimnames = list(rownames(df), colnames(df))
  )
}
sparseCountTables <- lapply(countTables, convert_to_sparse)

_After this step, all initial visualisations go well (I plotted a counts_vs_genes plot, and violin plots of counts per cell, and genes detected in all cells). I then continue with normalization, and it doesn't return me any error messages, only this warning:_

**Warning: Default search for "data" layer in "RNA" assay yielded no results; utilizing "counts" layer instead.**

_But when I check the number of genes before normalization it returns:_

> dim(mergedSeurat[["RNA"]]@counts)
**Error: no slot of name "counts" for this object of class "Assay5"**

When I then normalize anyway it seems to run as I get output. 
> mergedSeurat <- NormalizeData(mergedSeurat, normalization.method = "LogNormalize", scale.factor = 10000)
Normalizing layer: counts.WT pure (H3K4me3)
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
(And it does the same for the other 8 plates)

But then when I try to get Variable Features I get this:
> mergedSeurat <- FindVariableFeatures(mergedSeurat, selection.method = "vst", nfeatures = 2000)
Finding variable features for layer counts.WT pure (H3K4me3)
Calculating gene variances
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Calculating feature variances of standardized and clipped values
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
<error/rlang_error>
**Error in `VariableFeatures()`:
! No variable features found
---
Backtrace:
    ▆
 1. ├─Seurat::FindVariableFeatures(...)
 2. └─Seurat:::FindVariableFeatures.Seurat(...)
 3.   ├─Seurat::FindVariableFeatures(...)
 4.   └─Seurat:::FindVariableFeatures.StdAssay(...)
 5.     ├─SeuratObject::VariableFeatures(...)
 6.     └─SeuratObject:::VariableFeatures.Assay5(object, nfeatures = nfeatures, method = key)
 7.       └─rlang::abort(message = msg)**
 
 _And this is strange, because in V4 I get a ton of variable features._ 
 
 Sorry for the very long sequence of issues, but I thought let's just be complete right away so not to waste anyone's time with things I already tried :')  If you have any ideas please let me know! 

Thanks :)
